GLSP-1009: Provide web-extension example

tortmayr opened the pull request:
- Refactor codebase to provide entry points for both "normal" node VSCode extensions and web VSCode extensions. (via "@eclipse-glsp/vscode-integration/browser")
- Add additonal webview-extension example that covers two usecases
  - Communication with a GLSPServer that is running in the same extension context
  - Communication with a GLSPServer that is running externally via websocket
- Add launch configuration for starting the web extension

martin-fleck-at commented on example/workflow/web-extension/src/server/main.ts:
```diff
@@ -0,0 +1,27 @@
+import { ContainerModule } from 'inversify';
+import { GModelVSCodeStorage } from './gmodel-vscode-model-storage';
+
+export function createServerModules(): ContainerModule[] {
```
Could you please explain why this function is in a separate file? In the regular extension case, it is not and I don't see the benefit of it. Especially the name `main.ts` seems to imply that this is somehow compiled separately but I couldn't find anything related to it.

martin-fleck-at commented on example/workflow/web-extension/src/workflow-extension.ts:
```diff
@@ -0,0 +1,89 @@
+              clientName: 'workflow',
+              connectionOptions: { webSocketAddress }
+          })
+        : new NodeGlspVscodeServer({
```
The name `NodeGlspVscodeServer` is a bit confusing.

martin-fleck-at reviewed (changes requested):
Great work, thank you very much Tobias! It works well, I just have a few questions about the structure. I also noticed that we skipped the Elk layouting here, so if that causes issues we should open a follow-up ticket for it.

tortmayr commented:
Thanks Martin, I have addressed your feedback.
The `NodeGlspVscodeServer` is now only used in the node context in web/brower its called `BrowserGlspVscodeServer`.
In addition, I have restructured the web extension server code into a single file `workflow-server.ts`
